source/defclass-star.lisp

```diff
@@ -206,21 +206,21 @@ If the slot is a boolean, it ensures the name is suffixed with \"?\"."
   (concatenate-symbol name #.(symbol-package :asdf)))
 
 (defun slot-type-maybe-inherited (initform slot-name definition superclasses)
-  (let ((all-parents-finalized-p
-          (and
-           (every (lambda (s) (find-class s nil)) superclasses)
-           (every #'closer-mop:class-finalized-p
-                  (mapcar (lambda (s) (find-class s nil)) superclasses)))))
-    (cond
-      ((and all-parents-finalized-p
-            (find slot-name (apply #'append (mapcar #'mopu:slot-names superclasses))))
-       (let ((parent (find (list slot-name) superclasses
-                           :key #'mopu:slot-names :test #'intersection)))
-         (when parent
-           (getf (mopu:slot-properties parent slot-name) :type))))
-      (all-parents-finalized-p
-       (funcall *type-inference* initform slot-name definition))
-      (t nil))))
+  (if (null superclasses)
```

Comment:
Maybe rather have this as a separate `cond` clause?

In any way, it's not that critical and it's safe to merge without it.

Reply:
There are only 2 branches, why would `cond` be helpful here?

Reply:
`cond` is there in the innermost block. Moving the `null` check into it would simplify the code a bit, although create slightly more compile-time computations. Should not be critical, though.